honeybadger/payload.py

```diff
@@ -33,11 +39,8 @@ def is_not_honeybadger_frame(frame):
 
     logger.debug(tb)
 
-    payload = {
-        'class': type(exception) is dict and exception['error_class'] or exception.__class__.__name__,
-        'message': type(exception) is dict and exception['error_message'] or str(exception),
-        'backtrace': [dict(number=f[1], file=_filename(f[0]), method=f[2], source=read_source(f)) for f in reversed(tb)]
-    }
+    payload = prepare_exception_payload(exception)
+    payload['causes'] = [prepare_exception_payload(exception.__cause__) ] if hasattr(exception, '__cause__') else []
```

Comment:
Could we also add recursive causes to this list? For example, check to see if `exception.__cause__` also has a `__cause__` attribute, and add it to the list if so (and do that for each cause, until there is no additional `__cause__`?

Reply:
I thought of handling it recursively but wasn't sure if nested `causes` is supported by the notices spec.
I'll update.

Reply:
Yep, the array can have as many causes as you want. They are displayed in order, so the most recent cause should be displayed first, and the oldest cause should be displayed last.

Reply:
All done!